### PR TITLE
Update LDC 0.17.x to LLVM 5.0

### PR DESCRIPTION
**Pull request by redstar**
This PR updates ltsmaster to use the upcoming LLVM 5.0 version.

I hope that I picked all relevant commits. It does not compile yet - that's something for the next evening.

### CODE REVIEW

File: gen/functions.cpp [hunk 1]
```diff
@@ -391,13 +393,62 @@ void DtoResolveFunction(FuncDeclaration *fdecl) {
...
+                 lessPreciseFPMADOption ? "true" : "false");
+  func.addFnAttr("no-infs-fp-math", TO.NoInfsFPMath ? "true" : "false");
+  func.addFnAttr("no-nans-fp-math", TO.NoNaNsFPMath ? "true" : "false");
+#if LDC_LLVM_VER < 307
+  func.addFnAttr("use-soft-float", TO.UseSoftFloat ? "true" : "false");
+#endif
+
+  // Frame pointer elimination
+  func.addFnAttr("no-frame-pointer-elim",
+                 opts::disableFpElim ? "true" : "false");
```
Comment (thewilsonator) [inline gen/functions.cpp]: CI fails because "‘disableFpElim’ is not a member of ‘opts’". I think you forgot to cherrypick that too?

Comment (redstar) [inline gen/functions.cpp]: Yes, I am aware of this. It's only part of the problem. I think some files I picked are not needed (PGO for 0.17.x??). In general, it does not look too bad. I hope to fix it soon.

**Comment by JohanEngelen** [comment]
There are a bunch of things that are not needed: for example LTO things, `createAndSetDiagnosticsOutputFile` and `archiver.cpp`.

File: gen/functions.cpp (hunk 1 above)
Comment (redstar) [inline gen/functions.cpp]: Turned out to be a missing include.... :-)

**Comment by JohanEngelen** [comment]
To make things a little easier to grasp, could you already merge the adding of harmless LLVM5.0 things like FileCheck and profile-rt?

**Comment by JohanEngelen** [comment]
Frankly @redstar , I think it is much much easier to not cherry pick, and only fix the things that are broken, while looking at how they are done in master.  The current state of this PR adds many new things to 0.17.x that are not needed and are likely to be half-broken this way.

**Comment by redstar** [comment]
@JohanEngelen You can be right. I think I'll give it a try to fix the compile errors only. Nevertheless I committed the first profile-rt update as this is indeed harmless.

**Comment by JohanEngelen** [comment]
@redstar I've made my own branch, can we close this one?
https://github.com/ldc-developers/ldc/pull/2254

**Comment by redstar** [comment]
@JohanEngelen For sure. Thanks for the work.

**Pull request closed without merging**